source/client_channel_handler.c

```diff
@@ -219,9 +249,12 @@ static int s_packet_handler_connack(
     return AWS_OP_SUCCESS;
 }
 
-static int s_packet_handler_publish(
-    struct aws_mqtt_client_connection_311_impl *connection,
-    struct aws_byte_cursor message_cursor) {
+static int s_packet_handler_publish(struct aws_byte_cursor message_cursor, void *user_data) {
+    struct aws_mqtt_client_connection_311_impl *connection = user_data;
+    AWS_LOGF_TRACE(AWS_LS_MQTT_CLIENT, "id=%p: PUBLISH received", (void *)connection);
```

Comment:
Trivial: Should we log reception of PUBLISH at a debug level?

Reply:
I was going to add the log level refactor into this but later reconsidered.  But yes, that one should be debug.